### PR TITLE
chore: Updating tests which will fail on app invites getting merged

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

> Updating tests which will fail when app invites are merged on EE. Along with that, updating permission check for the invite modal content on app level.

Fixes [#21952](https://github.com/appsmithorg/appsmith/issues/21952)

## Type of change

- Chore (housekeeping or task changes that don't impact user perception)

- Manual
- Cypress

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by dipyamanbiswas07 — APPROVED** [review]
FE changes look fine to me.

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/MemberRoles_Spec.ts [hunk 1]
```diff
@@ -126,17 +128,18 @@ describe("Create new workspace and invite user & validate all roles", () => {
...
-    _.agHelper.Sleep();
-    _.agHelper.GetNClick(HomePage.selectRole);
+    cy.xpath(HomePage.selectRole).click();
     cy.get(".t--dropdown-option")
       .should("have.length", 3)
       .should("contain.text", `App Viewer`, `Developer`);
     cy.get(".t--dropdown-option").should("contain.text", `Administrator`);
-    _.agHelper.GetNClick(HomePage.editModeInviteModalCloseBtn);
+    _.agHelper.GetNClick(HomePage.closeBtn);
+    cy.get(_.homePage._appHoverIcon("edit")).first().click({ force: true });
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/MemberRoles_Spec.ts:141]: Why cant we use -     `_.agHelper.GetNClick(_.homePage._appHoverIcon("edit"));` instead.

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/MemberRoles_Spec.ts [hunk 2]
```diff
@@ -126,17 +128,18 @@ describe("Create new workspace and invite user & validate all roles", () => {
...
-    _.agHelper.GetNClick(_.homePage._appHoverIcon("edit"));
-    // cy.xpath(_.homePage._editPageLanding).should("exist");
+    cy.get(_.homePage._appHoverIcon("edit")).should("exist");
+
+    _.agHelper.GetNClick(_.homePage._shareWorkspace(workspaceId));
     _.agHelper.Sleep(2000);
-    _.agHelper.ClickButton("SHARE");
-    _.agHelper.Sleep();
-    _.agHelper.GetNClick(HomePage.selectRole);
+    cy.xpath(HomePage.selectRole).click();
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/MemberRoles_Spec.ts:135]: Why cant we use - `_.agHelper.GetNClick(HomePage.selectRole);` instead?

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/MemberRoles_Spec.ts [hunk 3]
```diff
@@ -126,17 +128,18 @@ describe("Create new workspace and invite user & validate all roles", () => {
     _.agHelper.Sleep();
     _.homePage.FilterApplication(appid, workspaceId);
     cy.get(_.homePage._applicationCard).first().trigger("mouseover");
-    _.agHelper.GetNClick(_.homePage._appHoverIcon("edit"));
-    // cy.xpath(_.homePage._editPageLanding).should("exist");
+    cy.get(_.homePage._appHoverIcon("edit")).should("exist");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/MemberRoles_Spec.ts:131]: `agHelper.AssertElementExist()` can be used here.

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/OtherUIFeatures/ExportApplication_spec.js [hunk 4]
```diff
@@ -104,14 +109,31 @@ describe("Export application as a JSON file", function () {
...
+        HomePage.InviteUserToWorkspaceFromApp(
+          Cypress.env("TESTUSERNAME1"),
+          "Developer",
+        );
+      } else {
+        HomePage.InviteUserToApplicationFromApp(
+          Cypress.env("TESTUSERNAME1"),
+          "Developer",
+        );
+      }
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/OtherUIFeatures/ExportApplication_spec.js:122]: Seeing this repeat a lot. Can we create a method in HomePage.ts helper & call it accordinlgy with respetcive role?

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/MemberRoles_Spec.ts [hunk 5]
```diff
@@ -157,6 +160,7 @@ describe("Create new workspace and invite user & validate all roles", () => {
       expect($list.eq(1)).to.contain(Cypress.env("TESTUSERNAME1"));
       expect($list.eq(2)).to.contain(Cypress.env("TESTUSERNAME2"));
     });
+    cy.get("[nam='arrow-right-s-fill']").should("not.exist");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/MemberRoles_Spec.ts:163]: `agHelper.AssertElementAbsence()` can be used.

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/OtherUIFeatures/ExportApplication_spec.js [hunk 6]
```diff
@@ -109,31 +110,14 @@ describe("Export application as a JSON file", function () {
...
-          Cypress.env("TESTUSERNAME1"),
-          Cypress.env("TESTPASSWORD1"),
-        );
-      } else {
-        cy.LoginUser(
-          Cypress.env("TESTUSERNAME1"),
-          Cypress.env("TESTPASSWORD1"),
-        );
-      }
+      cy.LogintoApp(Cypress.env("TESTUSERNAME1"), Cypress.env("TESTPASSWORD1"));
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/OtherUIFeatures/ExportApplication_spec.js]: LogintoApp() also does a Logout. Here is another logout before that. Are the cases running fine after new change?

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/OtherUIFeatures/ExportApplication_spec.js [hunk 7]
```diff
@@ -176,30 +156,13 @@ describe("Export application as a JSON file", function () {
...
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/OtherUIFeatures/ExportApplication_spec.js]: Duplicate LogOut().

Comment (ankitakinger) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/OtherUIFeatures/ExportApplication_spec.js]: I haven't added the `cy.LogOut` in this PR actually. It already existed. Should be running fine I think as it was running fine earlier.

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/OtherUIFeatures/ExportApplication_spec.js (hunk 6 above)
Comment (ankitakinger) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/OtherUIFeatures/ExportApplication_spec.js]: I haven't added the cy.LogOut in this PR actually. It already existed. Should be running fine I think as it was running fine earlier.

Comment (Aishwarya-U-R) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/OtherUIFeatures/ExportApplication_spec.js]: Even if its running - we can still remove it as it seems duplicate & make that change in this PR only. Other changes look OK!

**Review by dipyamanbiswas07 — APPROVED** [review]

File: app/client/src/ce/constants/workspaceConstants.ts [hunk 8]
```diff
@@ -21,7 +21,7 @@ export type WorkspaceUserRoles = {
   id: string;
   name: string;
   description: string;
-  entityType: string;
+  entityType: ENTITY_TYPE;
```
Comment (rishabhrathod01) [inline app/client/src/ce/constants/workspaceConstants.ts:24]: @ankitakinger are we using the entityType key in the backend? asking as in the context of evaluation `entityType` has a different meaning and we already have `ENTITY_TYPE` for it in frontend code, the below ENTITY_TYPE is the 3rd enum we have in our code base and this makes it error prone for future.

Comment (ankitakinger) [reply app/client/src/ce/constants/workspaceConstants.ts:24]: Its intentionally a different ENTITY_TYPE which is also used on backend for app invites.

**Pull request merged**